packages/flutter/lib/src/widgets/fade_in_image.dart

```diff
@@ -562,7 +562,7 @@ class _AnimatedFadeOutFadeInState extends ImplicitlyAnimatedWidgetState<_Animate
 
   @override
   Widget build(BuildContext context) {
-    if (widget.wasSynchronouslyLoaded || _placeholderOpacityAnimation!.isCompleted) {
+    if (widget.wasSynchronouslyLoaded || (_placeholderOpacityAnimation?.isCompleted ?? true)) {
```

Comment:
```suggestion
    if (widget.wasSynchronouslyLoaded ||
        (_placeholderOpacityAnimation?.isCompleted ?? true)) {
```

Reply:
should it instead be this? (`||` is on the second line)
```suggestion
    if (widget.wasSynchronouslyLoaded
        || (_placeholderOpacityAnimation?.isCompleted ?? true)) {
```